### PR TITLE
Feature/updated api gw uri

### PR DESCRIPTION
**Pull request by glow-mdsol**
Updates to include the amended pattern for API GW usage (code and examples)
Refactors some modules
Fixed remnants of pythonic naming
Increased Test coverage
Use built in Signing Algo (thanks @johnduhart )

@jcarres-mdsol FYI

### CODE REVIEW

**Comment by jcarres-mdsol** [comment]
@jfeltesse-mdsol may know some Golang himself also

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,2 +1,9 @@
 ## Version 1.0.1
 * Added `-pretty` as an argument to the tool to pretty print JSON output
+
+## Version 1.0.2
```
Comment (jfeltesse-mdsol) [inline CHANGELOG.md]: [nits] usually changelog go order desc, the latest changes are the most relevant thus should be on top.

File: VERSION [hunk 2]
```diff
@@ -1 +1 @@
-1.0.1
\ No newline at end of file
+1.0.2
```
Comment (jfeltesse-mdsol) [inline VERSION]: missing new lines, need to configure your editor?
see https://mdsol.jiveon.com/people/ssteeg@mdsol.com/blog/2018/11/06/no-more-no-newline-at-end-of-file-warnings

File: examples/medidata_apis/medidata_api_example.go [hunk 3]
```diff
@@ -29,7 +29,10 @@ func createClient() (mauthClient *go_mauth_client.MAuthClient, err error) {
 // Get the results from the Countries API
 func GetCountriesService(mauthClient *go_mauth_client.MAuthClient) (result *http.Response, err error) {
 	// Access the Countries API
-	targetUrl := "https://api.mdsol.com/v1/countries"
+	targetUrl := "https://api.mdsol.com/countries"
+
+	// make sure you specify the correct version
+	mauthClient.SetHeader("Mcc-version", "v2019-03-22")
```
Comment (jfeltesse-mdsol) [inline examples/medidata_apis/medidata_api_example.go]: [nits] headers are case insensitive but here and there where we mention this header we spell it "Mcc-Version"

File: CHANGELOG.md (hunk 1 above)
Comment (glow-mdsol) [reply CHANGELOG.md]: Ok, updated

File: VERSION (hunk 2 above)
Comment (glow-mdsol) [reply VERSION]: This is used by the go generate command, I think it doesn't want a newline

File: examples/medidata_apis/medidata_api_example.go (hunk 3 above)
Comment (glow-mdsol) [reply examples/medidata_apis/medidata_api_example.go]: Ok, updated.  I copied the text from the email asking for the changes

**Pull request merged**